### ISSUE TITLE
Scale lumisection by filter efficiency

### ISSUE THREAD

**Issue by bbockelm** [dmwm/WMCore#8510]
TaskChains can produce very small lumisections when many of the generated events are filtered out.

For example, if 99% of events are removed, a 300 event lumisection at the generator step becomes a 3 event lumisection for DIGI/RECO; the latter has seriously bad CPU efficiencies as we must serialize on lumi boundaries.

After discussion on the CPU efficiency mailing list, I think we've settled on tackling this at the computing level, not within the framework.

In particular, @ticoann and @vlimant suggested that we could apply the scaling here:

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/StdSpecs/TaskChain.py#L490-L496

I would suggest only applying the scaling if the input filtered efficiency is a sane value (non-negative, not greater than 100%, etc).

This does imply that we will depend somewhat on upstream to provide sane values for request efficiency.

**Comment by vlimant** [comment]
n.b. we already rely heavily on the filter-efficiency, which values is somehow more sane than you might think.
I am surprised of the absence of the /=efficiency at the line you point out to, but then, on the injection side of things:

https://github.com/cms-PdmV/cmsPdmV/blob/14e47fae465857eae9c5ad976781afbe2ac8ed41/mcm/json_layer/request.py#L2741
https://github.com/cms-PdmV/cmsPdmV/blob/14e47fae465857eae9c5ad976781afbe2ac8ed41/mcm/json_layer/request.py#L2834

this is already taken into account (probably a legacy work-around the fact that it is not taken into account in reqmgr) ; so any change to this needs to be carefully coordinated with mcm code.

**Comment by vlimant** [comment]
remembering this #6857

**Comment by amaltaro** [comment]
Do we want to update it in WMCore then? We can make this change for the next CMSWEB release if agreed.

**Comment by ticoann** [comment]
Is this still a issue? Real issue is #6857, right? In that case, we can

**Comment by vlimant** [comment]
we need to hold on for the agreement on doing the coordinated changes on both ends mcm/reqmgr

**Comment by vlimant** [comment]
@anorkus has agreed to push the change on mcm when the change is deployed for reqmgr.
@amaltaro please go ahead with a change, and coordinate deployment

**Comment by amaltaro** [comment]
Thanks! Antanas, let's plan this change for *May* production upgrade.

**Comment by amaltaro** [comment]
Before I forgot, we still have to figure out what's the best way to implement it. The best would be to "recalculate" EventsPerLumi according to the FilterEfficiency and update EventsPerLumi with the new value. Problem is: the original value would be lost and Ops clone/resubmission would retrigger the EventsPerLumi calculation...

We might add another argument for the original values and not change those... let's see.

**Comment by bbockelm** [comment]
This is marked as milestone WMAgent1804.  Is that correct?

**Comment by ticoann** [comment]
Yes.  I think we haven concluded how to implement this. (If we want to move this in WMCore instead of McM unified handling this)
1. Keep EventsPerLumi as original value without filter efficiency applied (and change the current code to apply only internally)
2. Change EventsPerLumi value by applying filter efficiency (but in that case we have to keep original value in other parameter (i.e. EventsPerLumiWithoutFilterEffiency) (can't use the same parameter for 2 different meaning if we change the value internally) 

Anyway I will move the milestone to this month and lower the priority since currently this is not broken.

**Comment by amaltaro** [comment]
Pushing it to the middle of 2019, but we might simply accept it's not ideal and let McM keep taking care of it.

**Comment by bbockelm** [comment]
@amaltaro - is this a potential project for @khurtado to tackle?

**Comment by amaltaro** [comment]
This is a tricky issue and we haven't converged on the exact implementation yet, so maybe not (yet).
